worker/src/unstract/worker/clients/test_docker.py

```diff
@@ -0,0 +1,195 @@
+        repository="test-image",
+        tag="latest",
+        stream=True,
+        decode=True,
+    )
+
+
+def test_get_container_run_config(docker_client, mocker):
+    """Test the get_container_run_config method."""
+    os.environ[Env.WORKFLOW_DATA_DIR] = "/source"
+    command = ["echo", "hello"]
+    organization_id = "org123"
+    workflow_id = "wf123"
+    execution_id = "ex123"
+
+    mocker.patch.object(docker_client, "_Client__image_exists", return_value=True)
+    mocker_normalize = mocker.patch(
+        f"{DOCKER_MODULE}.ContainerClientHelper.normalize_container_name",
+        return_value="test-image",
+    )
+    config = docker_client.get_container_run_config(
+        command,
+        organization_id,
+        workflow_id,
+        execution_id,
+        envs={"KEY": "VALUE"},
+        auto_remove=True,
+    )
+
+    mocker_normalize.assert_called_once_with("test-image")
+    assert config["name"] == "test-image"
+    assert config["image"] == "test-image:latest"
+    assert config["command"] == ["echo", "hello"]
+    assert config["environment"] == {"KEY": "VALUE"}
+    assert config["mounts"] == [
+        {
+            "type": "bind",
+            "source": f"/source/{organization_id}/{workflow_id}/{execution_id}",
+            "target": "/data",
+        }
+    ]
+
+
+def test_get_container_run_config_without_mount(docker_client, mocker):
+    """Test the get_container_run_config method."""
+    os.environ[Env.WORKFLOW_DATA_DIR] = "/source"
+    command = ["echo", "hello"]
+
+    mocker.patch.object(docker_client, "_Client__image_exists", return_value=True)
+    mocker_normalize = mocker.patch(
+        f"{DOCKER_MODULE}.ContainerClientHelper.normalize_container_name",
+        return_value="test-image",
+    )
+    config = docker_client.get_container_run_config(
+        command,
+        None,
+        None,
+        None,
+        auto_remove=True,
+    )
+
+    mocker_normalize.assert_called_once_with("test-image")
+    assert config["name"] == "test-image"
+    assert config["image"] == "test-image:latest"
+    assert config["command"] == ["echo", "hello"]
+    assert config["environment"] == {}
+    assert config["mounts"] == []
+
+
+def test_run_container(docker_client, mocker):
+    """Test the run_container method."""
+    # Patch the client object to control its behavior
+    mock_client = mocker.patch.object(docker_client, "client")
+
+    config = {
+        "name": "test-image",
+        "image": "test-image:latest",
+        "command": ["echo", "hello"],
+        "detach": True,
+        "stream": True,
+        "auto_remove": True,
+        "environment": {"KEY": "VALUE"},
+        "stderr": True,
+        "stdout": True,
+        "network": "",
+        "mounts": [],
+    }
+
+    assert isinstance(docker_client.run_container(config), DockerContainer)
+    mock_client.containers.run.assert_called_once_with(**config)
+
+
+if __name__ == "__main__":
+    pytest.main()
```

Comment:
@ritwik-g is this required while running with pytest? Doesn't it automatically pick up tests?

Reply:
Yes it does. But I guess if we ever run this particular file directly it will run test cases due to this.